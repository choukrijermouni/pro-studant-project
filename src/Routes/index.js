import React, { useEffect } from 'react'
import { Route, Switch } from 'react-router-dom'
import { ConnectedRouter } from 'connected-react-router'
import Home from 'pages/Home'
import Admin from 'pages/Admin'
import Test from 'pages/Test'
import Learners from 'pages/Learners'
import LearnerProfile from 'pages/LearnerProfile'
import LearnerHistory from 'components/Learner/LearnerHistory'
import LearnerAffectationHistory from 'components/Learner/LearnerAffectationHistory'
import Teams from 'pages/Teams'
import MyAccount from 'pages/MyAccount'
import { history } from 'store'
import Manager from 'pages/Manager'
import ManagerDetails from 'pages/ManagerDetails'
import TeamDetails from 'pages/TeamDetails'
import Signin from 'pages/Signin'
import Unauthorized from 'pages/Unauthorized'
import { useSelector, useDispatch } from 'react-redux'
import AdminProfile from 'components/Admin/AdminProfile'
import Error404 from 'pages/404'
import Generic404 from 'pages/Generic404'
import Signout from 'pages/Signout'
import { SignOutPath } from 'identity'
import { AdminRole, ManagerRole } from 'constants/'
import Reports from 'pages/Reports'
import InvitedProfile from 'components/Profile/InvitedProfile'
import InvitedManager from 'components/Manager/InvitedManager'
import { verifyInvitationAction } from 'pages/OnboardingContainer/store'
import { useQuery } from 'hooks'
import OnBoardingManager from 'pages/OnboardingContainer'

const UnauthenticatedRouter = _ => {
  const dispatch = useDispatch()
  const token = useQuery('token')
  useEffect(() => {
    if (token) {
      dispatch(verifyInvitationAction(token))
    }
  }, [token])
  return (
    <Switch>
      {token && <OnBoardingManager isInvited />}
      <Route exact path={SigninPath}>
        <Signin />
      </Route>
      <Route exact path={SignOutPath}>
        <Signout />
      </Route>
      <Route path={HomePath}>
        <Unauthorized />
      </Route>
      <Route path={generic404Path}>
        <Generic404 />
      </Route>
    </Switch>
  )
}

export default _ => {
  useEffect(() => {
    history.listen(() => { window.scrollTo(0, 0) })
  }, [])

  const { isConnected, user } = useSelector(({ identity }) => identity)
  const isAdminRole = user?.role?.includes(AdminRole)
  const isManagerRole = user?.role?.includes(ManagerRole)
  return (
        <ConnectedRouter history={history}>
          <Switch>
            <Route exact path={SigninPath}>
              <Signin />
            </Route>
            <Route exact path={SignOutPath}>
              <Signout />
            </Route>
            <Route exact path={HomePath}>
              <Home />
            </Route>
            <Route exact path={LearnerHistoryPath}>
              <LearnerHistory />
            </Route>
            <Route exact path={LearnerAffectationHistoryPath}>
              <LearnerAffectationHistory />
            </Route>
            <Route exact path={LearnersPath}>
              <Learners />
            </Route>
            <Route exact path={LearnerProfilePath}>
              <LearnerProfile />
            </Route>
            <Route exact path={InvitedProfilePath}>
              <InvitedProfile />
            </Route>
            <Route exact path={TeamsPath}>
              <Teams />
            </Route>
            <Route exact path={MyAccountPath}>
              <MyAccount />
            </Route>
            <Route exact path={AdminPath}>
              <Admin />
            </Route>
            <Route exact path={AdminProfilePath}>
              <AdminProfile />
            </Route>
            <Route exact path={TestPath}>
              <Test />
            </Route>
            <Route exact path={ManagersPath}>
              <Manager />
            </Route>
            <Route exact path={ManagerProfilePath}>
              <ManagerDetails />
            </Route>
            <Route exact path={InvitedManagerPath}>
              <InvitedManager />
            </Route>
            <Route exact path={TeamDetailsPath}>
              <TeamDetails />
            </Route>
            <Route exact path={RapportsPath}>
              <Reports />
            </Route>
            <Route path={pageNotFoundPath}>
              <Error404 />
            </Route>
            <Route>
              <Error404 />
            </Route>
          </Switch>
        </ConnectedRouter>)
}

export const ErrorPath = '/erreur'
export const HomePath = '/'
export const MyAccountPath = '/monCompte'
export const LearnersPath = '/apprenants'
export const LearnerProfileNeutralPath = '/apprenant'
export const LearnerProfilePath = `${LearnerProfileNeutralPath}/:id`
export const LearnerHistoryPath = `${LearnerProfileNeutralPath}/historique`
export const AffectationHistoryPath = `${LearnerProfileNeutralPath}/historiqueAffectation`
export const LearnerAffectationHistoryPath = `${AffectationHistoryPath}/:id`
export const NotificationsPath = '/notifications'
export const AdminPath = '/administrateur'
export const AdminProfileNeutralPath = `${AdminPath}/profile`
export const AdminProfilePath = `${AdminPath}/profile/:id`
export const TestPath = '/test'
export const ManagersPath = '/managers'
export const ManagerProfileNeutralPath = '/manager'
export const ManagerProfilePath = `${ManagerProfileNeutralPath}/:id`
export const TeamsPath = '/equipes'
export const TeamDetailsNeutralPath = '/equipe'
export const TeamDetailsPath = `${TeamDetailsNeutralPath}/:id`
export const SigninPath = '/signin'
export const pageNotFoundPath = '/404'
export const generic404Path = '/generic404'
export const CourseNeutralPath = '/tutoriel'
export const CertificatePath = '/Attestation'
export const RapportsPath = '/rapports'
export const InvitedProfilePath = '/apprenantInvite'
export const InvitedManagerPath = '/managerInvite'
