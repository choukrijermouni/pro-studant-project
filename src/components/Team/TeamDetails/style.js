import { blue, neutral } from '@pro_boa/ui'

const icon = {
  height: 30,
  width: 30
}
export default {
  cardsContainer: {
    display: 'flex',
    marginTop: 24,
    columnGap: 15
  },
  chartSection: {
    marginTop: 74
  },
  textClass: {
    paddingTop: 24
  },
  radarContainer: {
    width: '100%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  radarSubContainer: {
    width: 300
  },
  overlay: {
    width: 300,
    height: 300,
    position: 'absolute'
  },
  firstIcon: {
    position: 'absolute',
    top: 13,
    left: '23%',
    height: icon.height,
    width: icon.width
  },
  secondIcon: {
    position: 'absolute',
    top: 112,
    left: '-5%',
    height: icon.height,
    width: icon.width
  },
  thirdIcon: {
    position: 'absolute',
    top: 232,
    left: 8,
    height: icon.height,
    width: icon.width
  },
  fourthIcon: {
    position: 'absolute',
    top: 291,
    left: '44%',
    height: icon.height,
    width: icon.width
  },
  fifthIcon: {
    position: 'absolute',
    top: 232,
    right: '5%',
    height: icon.height,
    width: icon.width
  },
  sixthIcon: {
    position: 'absolute',
    top: 112,
    right: -10,
    height: icon.height,
    width: icon.width
  },
  seventhIcon: {
    position: 'absolute',
    top: 16,
    right: 63,
    height: icon.height,
    width: icon.width
  },
  icon: {
    cursor: 'pointer',
    fontSize: 20,
    color: blue[0],
    transition: 'transform 0.2s ease'
  },
  iconContainer: {
    marginTop: 42,
    display: 'flex',
    alignItems: 'center'
  },
  selectList: {
    width: '100%'
  },
  licenseButton: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-around',
    marginLeft: 24,
    padding: 8,
    width: 140,
    cursor: 'pointer',
    height: 45,
    borderRadius: 4,
    backgroundColor: neutral[1]
  },
  teamTag: {
    width: 43,
    height: 43,
    borderRadius: '50%',
    marginLeft: 16
  },
  Bio: {
    display: 'flex',
    alignItems: 'center',
    marginTop: 24,
    overflow: 'auto',
    paddingBottom: 12,
    '&::-webkit-scrollbar': {
      height: 5
    },
    '&::-webkit-scrollbar-thumb': {
      background: blue[1]
    },
    overflowY: 'hidden'
  },
  checkBox: {
    width: '120px !important'
  },
  cardContent: {
    display: 'flex'
  },
  leftSide: {
    width: 300
  },
  circle: {
    border: '2px solid #3767DA',
    width: 12,
    height: 12,
    backgroundColor: 'rgba(55,103,218,0.2)',
    borderRadius: 15
  },
  circle1: {
    border: '2px solid #F7B500',
    width: 12,
    height: 12,
    backgroundColor: 'rgba(247,181,0,0.25)',
    borderRadius: 15
  },
  flexBox: {
    display: 'flex',
    flexDirection: 'column',
    marginTop: 9
  },
  userContainer: {
    cursor: 'pointer'
  },
  searchBar: {
    width: 300
  },
  selectItem: {
    width: '34%',
    minWidth: 165
  },
  filterStyle: {
    width: '20%',
    minWidth: 150
  },
  firstPart: {
    width: '48%',
    display: 'flex',
    alignItems: 'center'
  },
  secondPart: {
    width: '48%',
    columnGap: 11,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end'
  },
  cardStyle: {
    height: 45,
    padding: '0 14px',
    whiteSpace: 'nowrap',
    borderRadius: 4,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: `${blue[0]}2D`,
    cursor: 'pointer',
    '&:hover > $icon': {
      transform: 'translateY(-3px)'
    },
    '&:hover': {
      backgroundColor: `${blue[1]}2D`
    }
  },
  headerContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    padding: '48px 0 24px',
    minWidth: 900
  },
  BioStyle: {
    display: 'flex',
    cursor: 'pointer',
    alignItems: 'center'
  },
  avatarContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    marginLeft: 24
  },
  paperDate: {
    backgroundColor: neutral[0],
    padding: '8px 12px',
    alignItems: 'center',
    cursor: 'pointer'
  },
  containerActions: {
    display: 'flex',
    margin: '18px 0'
  },
  calendarIconClass: {
    fontSize: 18,
    color: blue[0],
    lineHeight: 0,
    marginLeft: 8
  },
  showMoreContainer: {
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
    marginBottom: 50
  },
  titleClass: {
    alignItems: 'center'
  },
  BioStyleNoManager: {
    display: 'flex',
    alignItems: 'center'
  },
  nameStyle: {
    whiteSpace: 'nowrap',
    cursor: 'default'
  },
  teamInfosSkeleton: {
    display: 'flex',
    alignItems: 'center',
    width: '450px',
    margin: '0 16px'
  },
  descriptionSkeleton: {
    display: 'flex',
    width: 200
  },
  managerSkeleton: {
    display: 'flex',
    width: 267,
    alignItems: 'flex-end'
  },
  userCardWrapper: {
    marginBottom: 30
  },
  avatarManagerSkeleton: {
    position: 'relative',
    top: 9
  },
  dotsPopoverContainer: {
    backgroundColor: 'rgba(24,59,86,0.04)',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
    marginLeft: 16,
    gap: 3,
    cursor: 'pointer',
    height: 38,
    width: 38,
    borderRadius: 4
  },
  dot: {
    width: 5,
    height: 5,
    borderRadius: '50%',
    backgroundColor: blue[0]
  },
  popOverContent: {
    padding: '5px 0',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'flex-start'
  },
  linkTextStyle: {
    cursor: 'pointer',
    '&:hover': {
      color: blue[0]
    }
  },
  disabledLinkTextStyle: {
    '&:hover': {
      cursor: 'default'
    }
  },
  teamNameStyle: {
    minWidth: 120
  },
  managerContainer: {
    marginLeft: 10,
    display: 'flex'
  },
  linkContainer: {
    padding: '5px 10px',
    cursor: 'pointer',
    width: '100%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    transition: 'background 0.3s ease',
    '& > p': {
      transition: 'color 0.3s ease'
    },
    '&:hover': {
      backgroundColor: neutral[1]
    },
    '&:hover > p': {
      color: blue[0]
    }
  },
  disabledLinkContainer: {
    padding: '5px 10px',
    cursor: 'default',
    width: '100%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between'
  }
}
