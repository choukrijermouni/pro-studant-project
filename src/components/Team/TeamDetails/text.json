{
  "duration": "Total de la durée de visionnage",
  "work": "Compétences",
  "number": "52",
  "hours": "Heures",
  "average": "Moyenne de 15 Heures par apprenant",
  "percent": "4%",
  "noLicences": "Des apprenants de votre organisme n’ont pas de licence.",
  "lastAff": "Dernières affectations",
  "licences": "licences",
  "percentActive": "87%",
  "numberOfLicences": "23",
  "actives": "des apprenants actifs dans votre organisme.",
  "lastConnection": "Dernière connexion",
  "app": "apprenants",
  "manager": "managers",
  "header": "Nouvelles formations sur pro_boa",
  "noLicence": "Apprenants sans licence",
  "finished": "Formations complétées",
  "compare": "Comparer à :",
  "subCategories": "Équipes :",
  "selectTeam": "Sélectionnez une équipe",
  "categorie": " Windows server 2019",
  "searchLabel": "Trouver une formation",
  "showMore": "Afficher plus",
  "titles": {
    "virtual": "Équipe virtualisation",
    "dev": "Équipe développement"
  },
  "download": "Exporter le rapport de suivis",
  "downloadLeanersRecap": "Exporter liste des apprenants",
  "CheckBox": "Tout sélectionner",
  "handleLicenses": "Gérer les licences",
  "rule": "Managers :",
  "plusText": {
    "changeTeam": "Modifier l'équipe",
    "detachFromTeam": "Retirer de l'équipe",
    "detachFromOrganism": "Retirer de l'organisme",
    "exportLearnersList": "Exporter la liste des apprenants",
    "authorizedFormation": "Formations autorisées"
  },
  "none": "Sans manager d'équipe",
  "noName": "Sans nom",
  "noLearners": "Aucun apprenant",
  "SubTitle": "Faites vos premiers pas et invitez vos apprenants pour construire vos équipes.",
  "Title": "Invitez vos apprenants rejoindre l’aventure !",
  "managerTitle": "Aucun apprenants dans votre équipe pour le moment",
  "managerSubTitle": "Contactez votre administrateur pour inviter les apprenants de votre équipe.",
  "description": "Description :",
  "name": "Nom complet",
  "creationDate": "Date de création",
  "lastConnectionDate": "Dernière connexion",
  "endDate": "Expiration de licence",
  "editTeam": "Modifier l'équipe",
  "assignManager": "Affecter un manager",
  "removeManager": "Retirer un manager",
  "deleteTeam": "Supprimer l'équipe"
}