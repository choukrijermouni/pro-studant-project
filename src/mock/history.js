export default [
  {
    accessType: 'Navigateur (Chrome)',
    place: 'France (196.00.000.134)',
    date: '05:33 (il y a 4 minutes)',
    time: '3 heures'
  },
  {
    accessType: 'Navigateur (Chrome)',
    place: 'France (196.00.000.134)',
    date: '05:33 (il y a 4 minutes)',
    time: '3 heures'
  },
  {
    accessType: 'Navigateur (Chrome)',
    place: 'France (196.00.000.134)',
    date: '05:33 (il y a 4 minutes)',
    time: '3 heures'
  },
  {
    accessType: 'Navigateur (Chrome)',
    place: 'France (196.00.000.134)',
    date: '05:33 (il y a 4 minutes)',
    time: '3 heures'
  },
  {
    accessType: 'Navigateur (Chrome)',
    place: 'France (196.00.000.134)',
    date: '05:33 (il y a 4 minutes)',
    time: '3 heures'
  }
]
