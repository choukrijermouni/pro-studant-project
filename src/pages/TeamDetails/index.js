import Layout from 'components/Common/Layout'
import { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useParams } from 'react-router-dom'
import {
  fetchTeamAction,
  fetchTeamLearnersAction,
  getManagersListAction,
  getTeamLearnersRecapAction,
  getTeamRecapAction
} from './store'
import {
  usePagination,
  useSort
} from '@pro_boa/ui'
import TeamDetails from 'components/Team/TeamDetails'

const sortField = 'CreationDate'

export default () => {
  const dispatch = useDispatch()
  const { id } = useParams()
  const { page, rowsPerPage, skip } = usePagination()
  const { asc, field } = useSort(sortField)
  const { loading } = useSelector(state => state.config)
  useEffect(() => {
    dispatch(fetchTeamAction(id))
    dispatch(getTeamRecapAction(id))
    dispatch(getTeamLearnersRecapAction(id))
    dispatch(getManagersListAction())
  }, [id])
  useEffect(() => {
    dispatch(fetchTeamLearnersAction(id, rowsPerPage, skip, asc, field))
  }, [id, rowsPerPage, skip, asc, field, page])
  return (
    <Layout menu='team'>
      <TeamDetails loading={loading} />
    </Layout>
  )
}
