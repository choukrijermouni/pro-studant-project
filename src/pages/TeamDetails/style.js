export default {
  chartSection: {
    marginTop: 74
  }
}
