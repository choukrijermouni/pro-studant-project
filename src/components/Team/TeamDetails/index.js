import { createUseStyles } from 'react-jss'
import style from './style'
import Text from './text.json'
import {
  TypographyElement,
  Icon,
  Icons,
  neutral,
  usePagination,
  useSort,
  red,
  Skeleton
} from '@pro_boa/ui'
import { useEffect, useState } from 'react'
import { useParams } from 'react-router-dom'
import { useDispatch, useSelector } from 'react-redux'
import { goBack, push } from 'connected-react-router'
import { InvitedProfilePath, LearnerProfileNeutralPath } from 'Routes'
import SearchBar from 'components/Common/SearchBar'
import { fetchTeamLearnersAction, setTeamLearnersPagination } from 'pages/TeamDetails/store'
import { amazonBucket, OrganizationTeamImages, totalViewQueryFields, AdminRole, helpLinks, defaultField } from 'constants/'
import { exportLearnersToExcel, exportTeamLearnersInfoToExcel } from 'helpers'
import ChartsSection from 'components/Common/ChartsSection'
import EmptyPage from 'components/Common/EmptyPage'
import { fetchInvitedLearnerInfosAction } from 'pages/Learners/store'
import DataTable from 'components/Common/DataTable'
import { CreationDateRow, EndDateRow, LastConnectionDateRow, NameRow } from 'pages/Learners/Rows'
import { Popover, PopoverContent, PopoverTrigger } from 'components/Common/PopOver'
import { useDrawer } from 'components/Common/Drawer/drawerContext'
import TeamDetailsSkeleton from '../TeamDetailsSkeleton'
import ChartsSectionSkeleton from 'components/Common/skeletons/ChartsSectionSkeleton'
import { sendNotificationAction, switchLastLicensePopupAction } from 'pages/LearnerProfile/store'
import LastLicenseModal from 'components/Common/Drawer/AffectLicense/LastLicenseModal'
import Pagination from 'components/Common/DataTable/Pagination'

const useStyle = createUseStyles(style)

export const Title = ({ text, style }) => {
  return (
    <div style={{ display: 'flex', alignItems: 'center' }}>
      <div className={style} />
      <TypographyElement
        variant='heading2'
        color={neutral[4]}
        fontWeight='bolder'
        fontSize='12px'
        lineHeight='20px'
        display='inline'
        spacing='1px 10px'
      >
        {text}
      </TypographyElement>
    </div>
  )
}

export default ({ loading }) => {
  const {
    icon,
    iconContainer,
    teamTag,
    Bio,
    searchBar,
    headerContainer,
    firstPart,
    secondPart,
    cardStyle,
    BioStyle,
    BioStyleNoManager,
    userCardWrapper,
    nameStyle,
    dotsPopoverContainer,
    dot,
    popOverContent,
    linkTextStyle,
    disabledLinkTextStyle,
    teamNameStyle,
    managerContainer,
    linkContainer,
    disabledLinkContainer,
    chartSection
  } = useStyle()
  const { id } = useParams()
  const dispatch = useDispatch()
  const { page, rowsPerPage, skip, setPage, setRowsPerPage } = usePagination()
  const { field, setField, asc, setAsc } = useSort(defaultField)
  const [search, setSearch] = useState('')
  const team = useSelector(state => state.teamDetails)
  useEffect(() => { setAsc(false) }, [])
  const { learners: teamLearners, LearnersCount: learnersCount, Description: description, managersList: allManagers, Managers: teamManagers } = useSelector(state => state.teamDetails)
  const { TeamLearningRecap = [] } = useSelector(state => state.teamDetails)
  const { teamLearnersRecap = [] } = useSelector(state => state.teamDetails)
  const { organizationLicenseTypes = {} } = useSelector(state => state.referential)
  const filteredTeamManagers = teamManagers?.filter(element => !element.InvitationId)
  const [isOpen, setIsOpen] = useState(false)
  const { emailTypes } = useSelector(({ referential }) => referential)
  const { user } = useSelector(({ identity }) => identity)
  useEffect(() => {
    dispatch(setTeamLearnersPagination(skip, rowsPerPage))
  }, [skip, rowsPerPage])
  useEffect(() => {
    dispatch(fetchTeamLearnersAction(id, rowsPerPage, skip, asc, field, search))
  }, [id, rowsPerPage, skip, asc, field, search, page])
  const { lastLicensePopupClosed } = useSelector(({ profile }) => profile)
  useEffect(() => {
    setIsOpen(!lastLicensePopupClosed)
  }, [lastLicensePopupClosed])
  const isAdmin = user?.role?.includes(AdminRole)
  const config = {
    handleRowClick: (data) => {
      if (data.InvitationId) {
        dispatch(fetchInvitedLearnerInfosAction(data))
        dispatch(push(InvitedProfilePath))
      } else dispatch(push(`${LearnerProfileNeutralPath}/${data.Id}`))
    },
    columns: [
      {
        property: 'FirstName',
        title: Text.name,
        sortable: true,
        render: (item, props, value) => {
          return (
            <NameRow value={value} />
          )
        }
      },
      {
        property: 'CreationDate',
        title: Text.creationDate,
        sortable: true,
        render: (item) => {
          return (
            <CreationDateRow item={item} />
          )
        }
      },
      {
        property: 'LastConnectionDate',
        title: Text.lastConnectionDate,
        sortable: true,
        render: (item) => {
          return (
            <LastConnectionDateRow item={item} />
          )
        }
      },
      {
        property: 'EndDate',
        title: Text.endDate,
        sortable: true,
        render: (item, props, value) => {
          return (
            <EndDateRow setPage={setPage} item={item} value={value} organizationLicenseTypes={organizationLicenseTypes} />
          )
        }
      }
    ]
  }
  const { openDrawer, closeDrawer } = useDrawer()
  const popOverContentMap = [
    {
      text: Text.editTeam,
      onHandleClick: () => openDrawer(
        {
          componentName: 'editTeam',
          props: {
            id,
            teamName: team?.Name,
            teamIcon: team?.Image,
            description: team?.Description,
            closeModal: closeDrawer
          }
        }
      )
    },
    {
      text: Text.assignManager,
      onHandleClick: () =>
        allManagers?.length && (filteredTeamManagers?.length !== allManagers?.length)
          ? openDrawer(
              {
                componentName: 'assignManager',
                props: {
                  id,
                  handleClose: closeDrawer
                }
              }
            )
          : null,
      className: allManagers?.length && (filteredTeamManagers?.length !== allManagers?.length) ? linkTextStyle : disabledLinkTextStyle,
      containerClassName: allManagers?.length && (filteredTeamManagers?.length !== allManagers?.length) ? linkContainer : disabledLinkContainer,
      color: allManagers?.length && (filteredTeamManagers?.length !== allManagers?.length) ? neutral[6] : neutral[2]
    },
    {
      text: Text.removeManager,
      onHandleClick: () =>
        filteredTeamManagers?.length
          ? openDrawer({
              componentName: 'removeManager',
              props: {
                id,
                handleClose: closeDrawer
              }
            })
          : null,
      className: filteredTeamManagers?.length ? linkTextStyle : disabledLinkTextStyle,
      containerClassName: filteredTeamManagers?.length ? linkContainer : disabledLinkContainer,
      color: filteredTeamManagers?.length ? neutral[6] : neutral[2]
    },
    {
      text: Text.deleteTeam,
      onHandleClick: () => openDrawer({
        componentName: 'removeTeam',
        props: {
          team,
          handleClose: closeDrawer
        }
      }),
      color: red[3]
    }
  ]
  return (
    <>
      {loading
        ? <TeamDetailsSkeleton />
        : (
          <>
            <LastLicenseModal
              openModal={isOpen}
              handleClose={() => {
                dispatch(sendNotificationAction(emailTypes.OrganizationLicensesDepleted))
                dispatch(switchLastLicensePopupAction(true))
                setIsOpen(false)
              }}
              handleSubmit={() => {
                dispatch(sendNotificationAction(emailTypes.OrganizationLicensesRequested))
                dispatch(switchLastLicensePopupAction(true))
                setIsOpen(false)
              }}
            />
            <div className={iconContainer}>
              <div onClick={() => dispatch(goBack())}><Icon iconName={Icons.roundedLeft} style={icon} /></div>
              <img src={team.Image ? `${team?.Image}` : `${amazonBucket.bucketBaseUrl}${OrganizationTeamImages.TeamIconPath}${OrganizationTeamImages?.Development}`} className={teamTag} />
              <TypographyElement
                dataTest='team-name'
                component='h2'
                variant='heading2'
                align='left'
                spacing='0 20px'
                color={neutral[6]}
                className={teamNameStyle}
              >
                {team?.Name ? team?.Name : Text.noName}
              </TypographyElement>
              {
                isAdmin
                  ? (
                    <>
                      <Popover>
                        <PopoverTrigger>
                          <div className={dotsPopoverContainer}>
                            <span className={dot} />
                            <span className={dot} />
                            <span className={dot} />
                          </div>
                        </PopoverTrigger>
                        <PopoverContent topPosition={40}>
                          <div className={popOverContent}>
                            {popOverContentMap.map((item, index) => (
                              <div
                                key={index}
                                className={item.containerClassName || linkContainer}
                                onClick={() => {
                                  item.onHandleClick()
                                }}
                              >
                                <TypographyElement
                                  key={index}
                                  color={item.color || neutral[6]}
                                  fontWeight='normal'
                                  fontSize='13px'
                                  lineHeight='16px'
                                  className={item.className || linkTextStyle}
                                >
                                  {item.text}
                                </TypographyElement>
                              </div>
                            ))}
                          </div>
                        </PopoverContent>
                      </Popover>
                    </>)
                  : null
              }
            </div>
            {
              description &&
              (
                <TypographyElement
                  component='p'
                  variant='body2'
                  align='left'
                  spacing='16px 0 0 35px'
                  className={nameStyle}
                  color={neutral[4]}
                >
                  {`${Text.description} ${description}`}
                </TypographyElement>)
            }
            {
              teamManagers?.length
                ? (
                  <div
                    className={Bio}
                  >
                    <TypographyElement
                      component='p'
                      variant='body2'
                      align='left'
                      spacing='0 0 0 35px'
                      className={nameStyle}
                      color={neutral[4]}
                    >
                      {Text.rule}
                    </TypographyElement>
                    <div className={managerContainer}>
                      {
                        teamManagers?.map((manager, index) => (
                          !manager.InvitationId &&
                          (
                            <div
                              data-test='manager-avatar'
                              key={manager.Id}
                              className={isAdmin ? BioStyle : BioStyleNoManager}
                            >
                              <TypographyElement
                                component='p'
                                variant='body1'
                                className={nameStyle}
                                align='left'
                                spacing='0'
                              >
                                {index ? ', ' : ''}{manager?.FirstName} {manager?.LastName}
                              </TypographyElement>
                            </div>)
                        ))
                      }
                    </div>
                  </div>)
                : (
                  <div
                    className={BioStyleNoManager}
                  >
                    <TypographyElement
                      component='p'
                      variant='body2'
                      align='left'
                      spacing='12px 0 0 35px'
                      className={nameStyle}
                      color={neutral[4]}
                    >
                      {Text.rule}
                      <TypographyElement
                        display='inline'
                        component='span'
                        variant='body1'
                        align='left'
                        spacing='0 0 0 10px'
                        color={red[2]}
                      >
                        {Text.none}
                      </TypographyElement>
                    </TypographyElement>
                  </div>)
            }
          </>)}
      {learnersCount
        ? (
          <>
            {loading
              ? (
                <div className={chartSection}>
                  <ChartsSectionSkeleton />
                </div>)
              : <ChartsSection id={id} field={totalViewQueryFields.team} infoLink={helpLinks.teamProfile} />}
            {loading
              ? (
                <div className={headerContainer}>
                  <div className={firstPart}>
                    <div className={searchBar}>
                      <Skeleton lines={1} height={45} width={300} />
                    </div>
                  </div>
                  <div className={secondPart}>
                    <Skeleton lines={1} height={45} width={300} />
                    <Skeleton lines={1} height={45} width={300} />
                  </div>
                </div>
                )
              : (
                <div className={headerContainer}>
                  <div className={firstPart}>
                    <div className={searchBar}>
                      <SearchBar
                        height={45}
                        width={300}
                        handleChange={(value) => {
                          setSearch(value)
                          setPage(0)
                        }}
                        value={search}
                      />
                    </div>
                  </div>
                  <div className={secondPart}>
                    <div
                      onClick={() => exportLearnersToExcel(TeamLearningRecap, team?.Name ? team?.Name : Text.noName)}
                      className={cardStyle}
                      data-test='export-recap-to-excel-button'
                    >
                      <Icon iconName={Icons.download} style={icon} />
                      <TypographyElement
                        component='p'
                        fontSize='13px'
                        lineHeight='16px'
                        spacing='0 0 0 16px'
                        color={neutral[6]}
                      >
                        {Text.download}
                      </TypographyElement>
                    </div>
                    <div
                      onClick={() => exportTeamLearnersInfoToExcel(teamLearnersRecap, team?.Name ? team?.Name : Text.noName, organizationLicenseTypes)}
                      className={cardStyle}
                      data-test='export-learners-to-excel-button'
                    >
                      <Icon iconName={Icons.download} style={icon} />
                      <TypographyElement
                        component='p'
                        fontSize='13px'
                        lineHeight='16px'
                        spacing='0 0 0 16px'
                        color={neutral[6]}
                      >
                        {Text.downloadLeanersRecap}
                      </TypographyElement>
                    </div>
                  </div>
                </div>)}
            <div className={userCardWrapper}>
              <DataTable
                loading={loading}
                data={teamLearners}
                config={config}
                selectedField={field}
                sort={asc}
                handleSort={(e) => {
                  setField(e)
                  setAsc(!asc)
                }}
              />
            </div>
            <Pagination
              data={teamLearners}
              setRowsPerPage={setRowsPerPage}
              count={learnersCount}
              page={page}
              skip={skip}
              rowsPerPage={rowsPerPage}
              setPage={setPage}
            />
          </>)
        : (
          <EmptyPage
            Title={isAdmin ? Text.Title : Text.managerTitle}
            SubTitle={isAdmin ? Text.SubTitle : Text.managerSubTitle}
            manager
          />)}
    </>
  )
}
