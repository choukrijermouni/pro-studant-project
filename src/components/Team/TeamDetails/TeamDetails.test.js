import { shallow, configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router, Route } from 'react-router-dom'
import { ReduxProvider } from '../../../test/ReduxProvider'
import { createMemoryHistory } from 'history'
import { TeamDetailsPath } from '../../Routes'
import TeamDetails from '.'

configure({ adapter: new Adapter() })

describe('>>> TeamDetails --- Shallow Render ManagerCard component', () => {
  const history = createMemoryHistory()
  const route = '/teams/1'
  history.push(route)
  let wrapper
  let tree

  beforeEach(() => {
    wrapper = shallow(
      <ReduxProvider>
        <Router history={history}>
          <Route exact path={TeamDetailsPath}>
            <TeamDetails />
          </Route>
        </Router>
      </ReduxProvider>
    )
    tree = renderer.create(
      <ReduxProvider>
        <Router history={history}>
          <Route exact path={TeamDetailsPath}>
            <TeamDetails />
          </Route>
        </Router>
      </ReduxProvider>
    ).toJSON()
  })

  it('+++ render correctly and match snapshot', () => {
    expect(tree).toMatchSnapshot()
  })

  it('+++ render the component', () => {
    expect(wrapper).toBeTruthy()
  })
})
