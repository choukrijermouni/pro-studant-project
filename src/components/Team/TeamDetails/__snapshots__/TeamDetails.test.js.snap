// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> TeamDetails --- Shallow Render ManagerCard component +++ render correctly and match snapshot 1`] = `null`;
